src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/string_view.hpp

```diff
@@ -14,101 +14,517 @@
-using ::core::basic_string_view;
-using ::core::string_view;
-
-}  // namespace stdx
-}  // namespace v_noabi
-}  // namespace bsoncxx
-
-#elif defined(BSONCXX_POLY_USE_BOOST)
-
-#include <boost/version.hpp>
+#include <algorithm>
+#include <cstddef>
+#include <ios>
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <utility>
 
-#if BOOST_VERSION >= 106100
+#include <bsoncxx/stdx/operators.hpp>
+#include <bsoncxx/stdx/type_traits.hpp>
 
-#include <boost/utility/string_view.hpp>
-
-namespace bsoncxx {
-namespace v_noabi {
-namespace stdx {
-
-using ::boost::basic_string_view;
-using ::boost::string_view;
-
-}  // namespace stdx
-}  // namespace v_noabi
-}  // namespace bsoncxx
+#include <bsoncxx/config/prelude.hpp>
 
-#else
+#ifdef __has_include
+#if __has_include(<version>)
+#include <version>
+#endif
+#endif
 
-#include <boost/utility/string_ref.hpp>
+#ifdef __cpp_lib_string_view
+#include <string_view>
+#endif
 
 namespace bsoncxx {
 namespace v_noabi {
 namespace stdx {
 
-template <typename charT, typename traits = std::char_traits<charT>>
-using basic_string_view = ::boost::basic_string_ref<charT, traits>;
-using string_view = ::boost::string_ref;
-
-}  // namespace stdx
-}  // namespace v_noabi
-}  // namespace bsoncxx
-
+/**
+ * @brief Implementation of std::string_view-like class template
+ */
+template <typename Char, typename Traits = std::char_traits<Char>>
+class basic_string_view : bsoncxx::detail::equality_operators, bsoncxx::detail::ordering_operators {
+   public:
+    // Pointer to (non-const) character type
+    using pointer = Char*;
+    // Pointer to const-character type
+    using const_pointer = const Char*;
+    // Type representing the size of a string
+    using size_type = std::size_t;
+    // Type representing the offset within a string
+    using difference_type = std::ptrdiff_t;
+    // The type of the string character
+    using value_type = Char;
+
+    // Constant sentinel value to represent an impossible/invalid string position
+    static constexpr size_type npos = static_cast<size_type>(-1);
+
+   private:
+    // Pointer to the beginning of the string being viewed
+    const_pointer _begin = nullptr;
+    // The size of the array that is being viewed via `_begin`
+    size_type _size = 0;
+
+   public:
+    using traits_type = Traits;
+    using reference = Char&;
+    using const_reference = const Char&;
+    using const_iterator = const_pointer;
+    using iterator = const_iterator;
+    using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+    using reverse_iterator = const_reverse_iterator;
+
+    /**
+     * @brief Default constructor. Constructs to an empty/null string view
+     */
+    constexpr basic_string_view() noexcept = default;
+    constexpr basic_string_view(const basic_string_view&) noexcept = default;
+    bsoncxx_cxx14_constexpr basic_string_view& operator=(const basic_string_view&) noexcept =
+        default;
+
+    /// Default copy/move/assign/destroy
```

Comment:
```suggestion

    constexpr basic_string_view(const basic_string_view&) noexcept = default;
    bsoncxx_cxx14_constexpr basic_string_view& operator=(const basic_string_view&) noexcept =
        default;
```
Fix(?) spacing by separating the copy special member functions from the default ctor + remove a stray comment.